File: JavaScript/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion){
    const match = new Map();
    
    for (let i=0; i<participant.length; i++) {
        p = participant[i];
        c = completion[i];
        
        match.set(p, (match.get(p) || 0) +1);
        match.set(c, (match.get(c) || 0) -1);
    }
    
    for (let [k,v] of match) {
        if (v > 0) return k;
    }
}

File: Java/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> match = new HashMap<>();
        for (int i=0; i<participant.length; i++) {
            match.put(participant[i], match.getOrDefault(participant[i], 0) + 1);
            if (i < completion.length) {
                match.put(completion[i], match.getOrDefault(completion[i], 0) - 1);
            }
        }
        
        for (Map.Entry<String, Integer> entry: match.entrySet()) {
            if (entry.getValue() > 0) {
                return entry.getKey();
            }
        }
        return answer;
    }
}